### DISCUSSION TITLE
Trying to get JED approval but fails test

### DISCUSSION THREAD

**Discussion by alan123489** [joomengine/Joomla-Component-Builder#698] [Extention Issues] [answered]
Hi

I've written a few components with JCB and I'm now trying to get them onto JED.

I've run the JED checker program on one of them and I get errors e.g.

/admin/controllers/installed_module.php
You've used encoding in this file. This is not an error, but a JED editor will have to review this file.

this occurs on over 20 scripts and seems to be down to base64_decode.

In short if you have the words base64 anywhere it fails the JED checker.

Is there a way I can adjust things in JCB to turn off the use of base64 globally?

thanks

Alan

**Comment by Llewellynvdm** [comment, marked as answer]
When you can justify the base64_decoding as a legitimate use-case, it will be allowed. This you can quote me on in your JED submission. 

There are a number of use cases in JCB, and think... JCB is listed in the JED. But all its base64 use cases are legitimate... and therefore excepted.

But JCB has features that the JED will not allow... and these feature was used by me in private client projects that do not care for the JED rules... So you need to check each of these implementations to make sure they are legitimate use cases.

What are some of he legitimate use cases:
- Storing code in the database as base64 instead of the actual PHP code.
- To build a return url (standard Joomla behavior)
- and any other harmless internal movement of data in the component.

Not allowed...
- The WHMCS licensing integration.
- The rapping of functions in base64 that call home with hidden values
- ...any kind of obfuscation that breaks the GNU GPL license agreement.

Should you get stuck in the JED on this issue in what you think is a legitimate use-case please let me know.

**Reply by alan123489** [reply]
Thanks I'll see what they say
